app/graphql/resolvers/samples_resolver.rb

```diff
@@ -10,11 +10,18 @@ class SamplesResolver < BaseResolver
              description: 'Optional group identifier to return list of samples for.',
              default_value: nil
 
-    def resolve(group_id:)
-      if group_id
-        group = IridaSchema.object_from_id(group_id, { expected_type: Group })
-        authorize!(group, to: :sample_listing?, with: GroupPolicy)
-        authorized_scope(Sample, type: :relation, as: :namespace_samples, scope_options: { namespace: group })
+    argument :filter, Types::SampleFilterType,
+             required: false,
+             description: 'Ransack filter',
+             default_value: nil
+
+    def resolve(group_id:, filter:)
+      if filter && group_id
+        get_by_group_id(group_id:).ransack(filter.to_h).result
+      elsif filter
+        Sample.ransack(filter.to_h).result
```

Comment:
**nitpick (blocking):** This should be using the same scope as in the else. Might make sense to make a private `authorized_samples` method that can take an optional `group_id` so that it simplifies the rest of the resolve.

Reply:
Fixed in https://github.com/phac-nml/irida-next/pull/723/commits/b1d64665f1e06d44daed9b802c6b8592c5f12068